guacamole-ext/src/test/java/org/apache/guacamole/properties/EnumGuacamolePropertyTest.java

```diff
@@ -0,0 +1,372 @@
+         * zooplankton.
+         *
+         * @see <a href="https://en.wikipedia.org/wiki/Sardine">Sardine (Wikipedia)</a>
+         */
+        @PropertyValue("sardine")
+        SARDINE
+
+    }
+
+    /**
+     * Example enum consisting of a small set of possible vegetables. None of
+     * the values of this enum are annotated with {@link PropertyValue}.
+     */
+    public static enum Vegetable {
+
+        /**
+         * Potatoes are starchy root vegetables native to the Americas. The
+         * tuber itself is edible, but other parts can be toxic.
+         *
+         * @see <a href="https://en.wikipedia.org/wiki/Potato">Potato (Wikipedia)</a>
+         */
+        POTATO,
+
+        /**
+         * Carrots are root vegetables, tapered in shape and generally orange
+         * in color.
+         *
+         * @see <a href="https://en.wikipedia.org/wiki/Carrot">Carrot (Wikipedia)</a>
+         */
+        CARROT
+
+    }
+
+    /**
+     * Example Guacamole property which parses String values as Fish constants.
+     */
+    private static final EnumGuacamoleProperty<Fish> FAVORITE_FISH = new EnumGuacamoleProperty<Fish>(Fish.class) {
+
+        @Override
+        public String getName() {
+            return "favorite-fish";
+        }
+
+    };
+
+    /**
+     * Verifies that EnumGuacamoleProperty correctly parses string values that
+     * are associated with their corresponding enum constants using the
+     * {@link PropertyValue} annotation.
+     *
+     * @throws GuacamoleException
+     *     If a valid test value is incorrectly recognized by parseValue() as
+     *     invalid.
+     */
+    @Test
+    public void testParseValue() throws GuacamoleException {
+        assertEquals(Fish.SALMON,   FAVORITE_FISH.parseValue("salmon"));
+        assertEquals(Fish.TROUT,    FAVORITE_FISH.parseValue("trout"));
+        assertEquals(Fish.MACKEREL, FAVORITE_FISH.parseValue("mackerel"));
+        assertEquals(Fish.TUNA,     FAVORITE_FISH.parseValue("tuna"));
+        assertEquals(Fish.SARDINE,  FAVORITE_FISH.parseValue("sardine"));
+    }
+
+    /**
+     * Verifies that the absence of a property value (null) is parsed by
+     * EnumGuacamoleProperty as the absence of an enum constant (also null).
+     *
+     * @throws GuacamoleException
+     *     If a valid test value is incorrectly recognized by parseValue() as
+     *     invalid.
+     */
+    @Test
+    public void testParseNullValue() throws GuacamoleException {
+        assertNull(FAVORITE_FISH.parseValue(null));
+    }
+
+    /**
+     * Verifies that GuacamoleException is thrown when attempting to parse an
+     * invalid value, and that the error message contains a sorted list of all
+     * allowed values.
+     */
+    @Test
+    public void testParseInvalidValue() {
+        try {
+            FAVORITE_FISH.parseValue("anchovy");
+            fail("Invalid EnumGuacamoleProperty values should fail to parse with an exception.");
```

Comment:
Suggested change:
`fail("It isn't possible for anchovies to be anyone's favorite fish.");

;-)

Reply:
True.